test/helpers/setupConnection.ts

```diff
@@ -4,80 +4,85 @@ import { CountryCode } from '../../src/models/stringTypes.js'
+  if (!inviteBase64) throw new Error(`Invitation for ${receiverUrl} was not found`)
 
-  // verify and validate a new connection - holder
-  const receive = await fetchPost(`${receiverUrl}/connection/new/receive-invitation`, {
+  // Counterparty receives the invitation
+  await fetchPost(`${receiverUrl}/connection/new/receive-invitation`, {
     invite: inviteBase64,
     action: 'createConnection',
   })
-  const [holderConnectionId] = (await receive.text()).match(uuidRegex) || []
-  if (!holderConnectionId) throw new Error(`Connection was not found`)
-
-  let retries = 30
 
-  while (retries) {
-    await delay(500)
+  // Confirm the local connection record has been created
+  let attempt = 0
+  let inviterConnectionIdAtReceiver: string | undefined
+  while (attempt < 10) {
+    attempt++
+    await delay(1000)
     const receiverConnection = await fetchGet(`${receiverUrl}/connection?search=DIGI`)
-    const [receiverConnectionStatus] = (await receiverConnection.text()).match('Verification Code') || []
-    if (receiverConnectionStatus) {
-      retries = 0
-    } else if (retries === 1 && !receiverConnectionStatus) {
-      throw new Error('timeout')
-    } else {
-      retries--
+    ;[inviterConnectionIdAtReceiver] = (await receiverConnection.text()).match(uuidRegex) || []
+    if (inviterConnectionIdAtReceiver) {
+      break
+    }
+    if (attempt === 10) {
+      throw new Error(`No connection ID after 10 retries at ${receiverUrl}`)
     }
   }
 
-  await fetchPost(`${receiverUrl}/connection/${holderConnectionId}/pin-submission`, {
+  // Submit the pin code at the receiver
+  await fetchPost(`${receiverUrl}/connection/${inviterConnectionIdAtReceiver}/pin-submission`, {
     action: 'submitPinCode',
     pin: inviterPin,
     stepCount: '3',
   })
 
+  // Get the responding pin code from the counterparty to the original inviter
   const receiverEmail = await checkEmails('admin@veritable.com')
   const receiverPin = await extractPin(receiverEmail.id)
+  if (!receiverPin) throw new Error(`PIN for ${inviterUrl} was not found.`)
 
+  // Get the connection id for the counterparty at the original inviter
   const connections = await fetchGet(`${inviterUrl}/connection?search=OFFSHORE`)
-  const [inviterConnectionId] = (await connections.text()).match(uuidRegex) || []
-  if (!receiverPin || !inviterConnectionId) {
-    throw new Error(`PIN ${receiverPin} or Connection ID ${inviterConnectionId} not found`)
-  }
+  const [receiverConnectionIdAtInviter] = (await connections.text()).match(uuidRegex) || []
+  if (!receiverConnectionIdAtInviter) throw new Error(`No connection ID at ${inviterUrl}`)
 
-  await fetchPost(`${inviterUrl}/connection/${inviterConnectionId}/pin-submission`, {
+  // Original inviter submits the pin code for the counterparty
+  await fetchPost(`${inviterUrl}/connection/${receiverConnectionIdAtInviter}/pin-submission`, {
     action: 'submitPinCode',
     pin: receiverPin,
-    stepCount: '2',
+    stepCount: '3',
```

Comment:
NB I don't know why this works...but it does seem to make to progress more reliably

Reply:
I'll change and play with it